analyzer/javacg-opal/src/main/java/eu/fasten/analyzer/javacgopal/data/analysis/OPALMethod.java

```diff
@@ -104,7 +107,7 @@ public static String getPackageName(final Type parameter) {
                 return slashToDot(Objects
                         .requireNonNull(getPackageName(parameter.asArrayType().componentType())));
             } else if (parameter.asObjectType().packageName().equals("")) {
-                return null;
+                return "";
```

Comment:
Do we then end up with a double slash in the URLs? So before it was "/null/ClassName", now it is "//ClassName"... or is this being handled through canonicalize?

Reply:
Yes, now it is "//ClassName". Canonicalize does not handle it. In fact, if there is no namespace canonicalize complains about it. That is why we now check before canonicalizing to see if there is not any namespace please don't try.

Reply:
Hmm, I do not like the `//` construct... I would argue that the correct version would be `/ClassName`. Is there an easy way to achieve this?

Reply:
It is not allowd by the FastenUri class if you write this: 
```
var v = FastenURI.create("/ClassName");
```
you would get `java.lang.IllegalArgumentException: Missing entity`

Reply:
I also just realized that even "//ClassName" cannot be correctly parsed by FastenUri. It does not cause any exception but It assumes that `ClassName` is the product name while it is the name of the class. What should we do about this? should we use some meaningful placeholder for the package name such as `/no.package/ClassName`? or should we create another URI specific for the classes?

Reply:
How about using `/ClassName` and fixing the code that extracts the package/product/class information?

Reply:
In general, I think we should revise the `FastenURI` at some point. We should improve our validation and I also think that it does not make sense to parse the whole thing on initialization. I would prefer storing the string representation and only parse part of it on demand, we can still cache information, once parsed. ... but this is a discussion for another issue.